### PR TITLE
Purchase Details Mark I: Shipping & Download cells

### PR DESCRIPTION
**Pull request by mindgraffiti**
This PR includes:
- the Shipping cell for simple and variable products
- the Downloads cell for simple and variable products

The next PR will include the purchase note cell.

## To test
- Choose a variety of products (simple, variable, and/or downloadable) from your store and create a new order. 
- Navigate to Orders > Order Number > Fulfill Order and select each type of product
- A grouped or affiliate product should not have any Purchase Details section
- a downloadable product will have the Downloads cell
- a normal product will have the Shipping cell

Update release notes:

- [ ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: WooCommerce/Classes/ViewModels/ProductDetailsViewModel.swift [hunk 1]
```diff
@@ -109,6 +139,27 @@ final class ProductDetailsViewModel {
...
+        // After refreshing the results controller,
+        // let's look up some product settings it holds.
+        weightUnit = lookupProductSettings(Keys.weightUnit)
+        dimensionUnit = lookupProductSettings(Keys.dimensionUnit)
+    }
+
+    /// Look up Product Settings
+    ///
+    func lookupProductSettings(_ settingID: String) -> String? {
+        for setting in productSettings where setting.settingID == settingID {
```
Comment (ctarda) [inline WooCommerce/Classes/ViewModels/ProductDetailsViewModel.swift]: I'm just curious, but do you have any particular reason to prefer this syntax over something like this?
```
return productSettings.filter({$0.settingID == settingID}).first
```
Seriously, it's just curiosity

Comment (mindgraffiti) [reply WooCommerce/Classes/ViewModels/ProductDetailsViewModel.swift]: That's so pretty! I forget that `filter` and `map` exist. Going to change this code

**Comment by mindgraffiti** [comment]
@ctarda 

✅ Codegen setting fixed. Thanks for the warning!
✅ Made my function more swift-y

As for the blank information in the Shipping and Download cells:
I thought about adding "Weight: --kg" instead of blank strings, but I think the last time I looked this up, WCAndroid makes a point to show nothing at all. I added a check that prevents the cell from getting created in the first place, if the info is blank, and prevents the section from getting created if there are no rows. Would you re-test that part please?

**Comment by ctarda** [comment]
The shipping and download cells are not displayed when there is no information available:




Notice how, in the second screenshot, which corresponds to a product without images, it seems like there might be some extra space at the top of the table (where the product image should be).

If that's an issue, I think we could attack it in a separate PR, what do you think? Other than that...

:shipit:

**Comment by mindgraffiti** [comment]
Thanks @ctarda! So at first I thought "no that's supposed to be there" because we purposely leave space for a badge (Private, Draft, etc). But then I looked again and the i2 designs don't discuss what to do for Published products that have no images. I have a meeting with @kyleaparker today, I'll ask her if it should be collapsed.

**Comment by kyleaparker** [comment]
>Notice how, in the second screenshot, which corresponds to a product without images, it seems like there might be some extra space at the top of the table (where the product image should be).

If there is a badge and no image it should look like this: 

with-badge

And with no badge or image:

no-badges

**Pull request merged**